core/object.go

```diff
@@ -875,6 +885,42 @@ func (p Proc) GetType() *Type {
 }
 
 func (p Proc) Hash() uint32 {
+	return p.Fn.Hash()
+}
+
+func (p ProcFn) Call(args []Object) Object {
+	return p(args)
+}
+
+func (p ProcFn) Compare(a, b Object) int {
```

Comment:
One more thing: since we are not exposing ProcFn as Joker type, I don't think it needs to implement Object interface...

Reply:
I've removed all but the two receivers to which `Proc` delegates its implementations: `Equals()` and `Hash()`.

Want me to pull those implementations into `Proc`'s and remove them as well?

Reply:
Yes, that would make more sense to me unless you have specific reasons to keep them in ProcFn.

Reply:
That's done, leaving only `(ProcFn) Call()`, which only `csv_native.go` uses.

Might be worth having that code wrap the `ProcFn` it builds in a `Proc` so the `(ProcFn) Call()` receiver can be eliminated as well -- will try that next.

Reply:
That builds and passes tests, but offhand I can't find any tests specific to `csv.joke`, so am not sure I haven't broken something here. Looking into to coming up with a decent test....

Reply:
Okay, that seems to work just fine.